packages/lean-imt/src/lean-imt.ts

```diff
@@ -233,6 +233,67 @@ export default class LeanIMT<N = bigint> {
         this._nodes[this.depth] = [node]
     }
 
+    /**
+     * Updates m leaves all at once.
```

Comment:
If you set `m = n`, the complexity as I stated it becomes `O(n)` since `log(n)-log(m) = 0`. I think it makes sense to add this case separately to the complexity statement, but maintaining the whole explanation with a smaller `m`.

That is, my proof is a generalization of the case `m=n`, so I don't see why we should take that out. Completely agree with mentioning that the algorithm is `O(n)`, though.

Reply:
Isn't the worst case the one when the nodes to update have as few ancestors as possible on all levels? In that case, time complexity would be quite similar to the one of the `update` function in a loop, i.e. O(n * log n).

Reply:
Ok, I also meant `n` as the subset of nodes to be updated (and not all leaves). We are on the same page 👍🏽

Reply:
Yes, but as the number of leaves to update increases, the condition of "having few common ancestors" starts being stronger and stronger, to the point where common ancestors start appearing inevitably. This makes that we never reach `n*log(n)` operations.

Reply:
Hey @ChinoCribioli could you update the `updateMany` function documentation with something similar to what you have at the beginning and move the time complexity analysis to a new issue called something like `UpdateMany time complexity` and add a comment with your proof so that we can discuss it there. Then, if you want, you could add it to the LeanIMT paper. What do you think?

Reply:
Done. I changed the documentation to a more minimalist description and created [this issue](https://github.com/privacy-scaling-explorations/zk-kit/issues/316) to discuss the deeper complexity analysis.

After the discussion is closed, you may reach out to me to add the whole analysis to the LeanIMT paper.

Reply:
Great! Thank you very much! 🙏